General.ruleset

```diff
@@ -0,0 +1,110 @@
+  </Rules>
+  <Rules AnalyzerId="SonarAnalyzer.CSharp" RuleNamespace="SonarAnalyzer.CSharp">
+    <Rule Id="S101" Action="None" />
+    <Rule Id="S103" Action="Warning" />
+    <Rule Id="S104" Action="Warning" />
+    <Rule Id="S105" Action="Warning" />
+    <Rule Id="S1067" Action="Warning" />
+    <Rule Id="S107" Action="Warning" />
+    <Rule Id="S1109" Action="Warning" />
+    <Rule Id="S1128" Action="None" />
+    <Rule Id="S113" Action="Warning" />
+    <Rule Id="S1135" Action="None" />
+    <Rule Id="S1145" Action="Warning" />
+    <Rule Id="S1147" Action="Warning" />
+    <Rule Id="S1151" Action="Warning" />
+    <Rule Id="S1192" Action="Warning" />
+    <Rule Id="S1227" Action="Warning" />
+    <Rule Id="S1244" Action="Warning" />
+    <Rule Id="S127" Action="Warning" />
+    <Rule Id="S131" Action="Warning" />
+    <Rule Id="S134" Action="Warning" />
+    <Rule Id="S138" Action="Warning" />
+    <Rule Id="S1541" Action="Warning" />
+    <Rule Id="S1696" Action="Warning" />
+    <Rule Id="S1858" Action="Warning" />
+    <Rule Id="S2187" Action="None" />
+    <Rule Id="S2302" Action="Warning" />
+    <Rule Id="S2357" Action="Warning" />
+    <Rule Id="S2387" Action="Warning" />
+    <Rule Id="S2583" Action="None" />
+    <Rule Id="S2699" Action="None" />
+    <Rule Id="S3052" Action="Warning" />
+    <Rule Id="S3235" Action="Warning" />
+    <Rule Id="S3240" Action="Warning" />
+    <Rule Id="S3253" Action="Warning" />
+    <Rule Id="S3254" Action="Warning" />
+    <Rule Id="S3257" Action="Warning" />
+    <Rule Id="S3358" Action="None" />
+    <Rule Id="S3717" Action="Warning" />
+    <Rule Id="S3881" Action="None" />
+    <Rule Id="S3962" Action="Warning" />
+    <Rule Id="S4004" Action="Warning" />
+    <Rule Id="S4023" Action="Warning" />
+    <Rule Id="S4035" Action="None" />
+    <Rule Id="S4041" Action="Warning" />
+    <Rule Id="S4047" Action="Warning" />
+    <Rule Id="S4052" Action="Warning" />
+    <Rule Id="S4261" Action="Warning" />
+    <Rule Id="S881" Action="Warning" />
+  </Rules>
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
+    <Rule Id="SA0001" Action="None" />
+    <Rule Id="SA1101" Action="None" />
+    <Rule Id="SA1200" Action="None" />
+    <Rule Id="SA1202" Action="None" />
+    <Rule Id="SA1208" Action="None" />
+    <Rule Id="SA1309" Action="None" />
+    <Rule Id="SA1404" Action="None" />
+    <Rule Id="SA1413" Action="None" />
+    <Rule Id="SA1501" Action="None" />
+    <Rule Id="SA1502" Action="None" />
+    <Rule Id="SA1503" Action="None" />
+    <Rule Id="SA1507" Action="None" />
+    <Rule Id="SA1516" Action="None" />
+    <Rule Id="SA1519" Action="None" />
+    <Rule Id="SA1600" Action="None" />
+    <Rule Id="SA1601" Action="None" />
+    <Rule Id="SA1602" Action="None" />
+    <Rule Id="SA1633" Action="None" />
+    <Rule Id="SA1649" Action="None" />
+    <Rule Id="SX1101" Action="Warning" />
+    <Rule Id="SX1309" Action="Warning" />
```

Comment:
There is no possibility to add some exceptional cases for rules. Only suppress each warning individually or disable the rule completely.

But I would not use `T` and `H` for field names, besides of who recommends this unusual naming. Naming at least as `texts`/`strings`/`stringLocalizer` and `html`/`htmlLocalizer` doesn't look bad for me.

Reply:
The thing is that these have been Orchard conventions for the past decade, and all code in Orchard itself, as well as in the Orchard ecosystem, uses them (so it would be strange for others to see something else in our code).

However, one workaround to still satisfy the private field with underscore rule would be to keep these localizers in `public IStringLocalizer T { get; }`. But I don't like the idea of introducing a property just for the sake of this.

Checking it out, the rule itself is not too complex: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/b6b9b0249a142132453873a677b7f09d7d65f34b/StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SX1309FieldNamesMustBeginWithUnderscore.cs So we could create our own analyzer package (see: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2265) and use a modified copy of this just to exclude these two items.

I realize this is out of scope of this project so I'll think about it.

Reply:
It seems that it's not possible to just include analyzers as code in the project and use them in the solution, you have to build a proper project (or actually, multiple) for them and package them up (and have the compiler SDK installed). So my initial idea of creating a copied analyzer with the localizers as exceptions won't work... I've disabled it then.